sirepo/job_supervisor.py

```diff
@@ -594,11 +621,13 @@ def _create_op(self, opName, req, **kwargs):
         self.ops.append(o)
         return o
 
-    def _get_max_run_secs(self, op_name, kind, run_mode):
+    def _get_max_run_secs(self, op_name, run_mode, req):
         if op_name in _UNTIMED_OPS or \
             (run_mode == sirepo.job.SBATCH and op_name == job.OP_RUN):
             return 0
-        t = cfg.max_hours[kind]
+        t = cfg.max_hours[req.kind]
+        if req.kind == job.PARALLEL and req.content.get('isPremiumUser'):
+            t = cfg.max_hours['parallel_premium']
         if op_name == sirepo.job.OP_ANALYSIS:
```

Comment:
I think if `op_name == sirepo.job.OP_ANALYSIS` should be before `if req.kind`, and that second if should be an `elif`. It made me pause to read this, because the code says OP_ANALYSIS takes precedence, but it's written in the opposite order.